### ISSUE TITLE
Select Next Occurrence: add support for `ToggleCaseSensitive` and `ToggleWholeWord`

### ISSUE THREAD

**Issue by vladmoroz** [zed-industries/zed#15415]
### Check for existing issues

- [X] Completed

### Describe the feature

In VS Code, Select Next Occurrence has the same modes as search, allowing to configure whether next selection is case-sensitive or grabs whole words:

https://github.com/user-attachments/assets/13af3ce5-179c-495b-bfc0-5891990b6823

This is currently missing in Zed. It's really helpful for refactors on the fly via Select Next Occurrence compared to search, which forces you to go either match by match, or to replace all matches altogether.

**Comment by AurevoirXavier** [comment]
This should be prioritized highly, in my opinion.

**Comment by AurevoirXavier** [comment]
It's truly a vital feature. Particularly, when the LSP is not performing correctly, such as with the RA nightly frequently crashing, we are unable to use F2 to rename the variable. The replace function will change all matches. Selecting the next occurrence is the best available option.

**Comment by brandonzylstra** [comment]
I agree this is EXTREMELY high priority.  Zed's "Select Next Occurrence" is almost useless until this bug/limitation is fixed.

**Comment by laymonage** [comment]
Was looking for this and expected it to work as it does on VSCode using the <kbd>Cmd+Option+C</kbd> and <kbd>Cmd+Option+W</kbd> shortcuts before pressing <kbd>Cmd+D</kbd>, but was very surprised that this is still unsupported ☹️...

**Comment by burns-brian** [comment]
I was reading a post by [dinocosta](https://github.com/zed-industries/zed/issues/41070#issuecomment-3451704051), and he says you can go to Settings > Search & Files > Toggle "Case Sensitive" On. This should make `cmd + d` only select same-case items.

**Comment by dinocosta** [comment]
Hey everyone 👋

Now that #64137 has been merged, I believe we're ready to close this discussion! 🎉

The `editor: select next`, `editor: select previous`, and `editor: select all matches` actions now respect both the "Match Case Sensitivity" and "Match Whole Words" options from the buffer search bar when the search bar is visible. When the search bar isn't visible, those actions use the `search.case_sensitive` and `search.whole_word` settings.

It's worth noting that, when starting from a single caret, whole-word matching is always enabled, similar to VS Code. Case sensitivity still follows the search bar, or your settings when the search bar is hidden. Changing either option while adding selections applies to the next action.

Since that covers the case-sensitive and whole-word support requested here, I'm going to close this discussion.

If you run into a case where this isn't working as expected, such as one of these options not being respected, please open a new issue with the steps to reproduce it and feel free to link to this discussion.

Thanks! 🙂
